### ISSUE TITLE
GPLayer's prediction seems to be too confident

### ISSUE THREAD

**Issue by dtchang** [secondmind-labs/GPflux#37]
Looking at the "Hybrid Deep GP models: ..." tutorial, the GPLayer's prediction seems to be too confident, i.e. its uncertainty estimate (95% confidence level) does not cover the training data spread. Its prediction accuracy (mean), however, is very good, nearly identical to that of the neural network model obtained by removing the GPLayer.

When I replace the GPLayer with a TFP DenseVariational layer using Gaussin priors, the prediction accuracy is not as good. However, importantly, the uncertainty estimate is very good, covering the training data spread well.

Without good uncertainty estimate, the GPLayer seems to add little value over the neural network model, which already provides good prediction accuracy.

**Comment by vdutor** [comment]
Hi @dtchang, I had another look at the notebook and I think the fit is actually pretty good. [In this PR](https://github.com/secondmind-labs/GPflux/pull/41), I updated the notebook so that it plots the mean and variance of `y` instead of `f`. In the figure below, you can see that the fit now encloses the data. You can also observe the increased uncertainty on `f` in the far left and right of the domain.

![fit_snelson_gpflux_issue](https://user-images.githubusercontent.com/13780846/128022460-e404f77c-ea96-43d7-9c31-7810a1652c43.png)

**Comment by dtchang** [comment]
Vincent,

If so, that will be great. I can’t find the updated notebook in the GitHub. Please send it to me or update the GitHub asap. Thanks.

Regards,
Dan

Sent from Mail for Windows 10

**Comment by dtchang** [comment]
I still can't find the updated notebook in the GitHub. But, I think this issue should be re-opened because it is not a plotting problem, but a prediction problem. f is the prediction, whereas y is the target. So, it seems to me the plot should be based on the mean and variance of f, not y, as in the published tutorial. BTW, the issue of poor uncertainty estimation exists not just in this tutorial, but also in other tutorials (e.g., "Introduction to GPflux" and "Why GPflux is a modern (deep) GP library").
